src/ModalHeader.js

```diff
@@ -4,24 +4,35 @@ import classnames from 'classnames';
 
 class ModalHeader extends React.Component {
 
+  renderCloseButton() {
+    if (this.props.closeButton === true) {
+      return (
+        <button
+          className='close'
+          aria-label={this.props['aria-label'] || 'Close'} //eslint-disable-line react/prop-types
+          onClick={this.props.onHide}
+          style={{ marginTop: -2 }}
+        >
+          <span aria-hidden="true">
+            &times;
+          </span>
+        </button>
+      );
+    }
+    else if (React.isValidElement(this.props.closeButton)) {
```

Comment:
I imagine `this.props.children` doesn't know about `this.props.onHide`? If it somehow does, I don't think this `else` would be necessary. If it doesn't, I'd pass down `this.props.onHide` as `onHide` and not as `onClick` for all we know that custom "close button" could be a drop-down with options.

Another way would be to sniff whether `this.props.children` is alone or there are many, and pass it (or to the first) the `onHide`, but I'm not a huge fan of that approach.